dynamite_nsm/services/kibana/package/manager.py

```diff
@@ -0,0 +1,784 @@
+            password (str): ES Auth password
+            force (bool): force uninstall from all spaces?
+        """
+        print(f"Preparing {len(packages)} for uninstall..")
+        for package in packages:
+            package.reload_installed_objects()
+            package.uninstall(self.kibana_url, auth=(username, password))
+
+    def import_kibana_saved_objects(self, username: str, password: str, kibana_objects_file: IO[AnyStr],
+                                    space: Optional[str] = None, overwrite: Optional[bool] = True,
+                                    create_copies: Optional[bool] = False):
+        """Import saved objects into kibana from a package file
+
+        Args:
+            username (str): kibana auth usrname
+            password (str): kibana auth passwd
+            kibana_objects_file (IO[AnyStr]): the file to parse and install
+            space (Optional[str], optional): id of the space to install the object to. Defaults to None.
+            overwrite (Optional[bool], optional): overwrite existing ids?. Defaults to True.
+            create_copies (Optional[bool], optional): create copies if an object exists with the same id?. Defaults to False.
+
+        Raises:
+            ValueError: Something went wrong
+
+        Returns:
+            dict: Response from kibana
+        """
+
+        auth = self._get_kibana_auth_securely(username, password)
+
+        if space:
+            url = f'{self.kibana_url}/s/{space}/api/saved_objects/_import'
+        else:
+            url = f'{self.kibana_url}/api/saved_objects/_import'
+        if all([overwrite, create_copies]):
+            raise ValueError("createNewCopies and overwrite cannot be used together.")
+
+        params = {'overwrite': overwrite, 'createNewCopies': create_copies}
+
+        # TODO: Catch connection denied when kibana is down and handle/inform user gracefully
+        if isinstance(kibana_objects_file, IOBase):
+            reqdata = {'file': ('dynamite_import.ndjson', kibana_objects_file)}
+        else:
+            reqdata = {'file': kibana_objects_file}
+        resp = requests.post(url, params=params, auth=auth, files=reqdata, headers={'kbn-xsrf': 'true'})
+        if resp.status_code not in range(200, 299):
+            self.logger.error(f'Kibana endpoint returned a {resp.status_code} - {resp.text}')
+            # TODO raise exception
+        return resp.json()
+
+    def install(self, package_install_path: str, username: Optional[str] = None, password: Optional[str] = None):
+        """Install a package. A package can be given as an archive or directory. A package must contain one or more ndjson files and a manifest.json
+
+        Args:
+            package_install_path (str): path to the file or folder of files
+            username (Optional[str], optional): kibana auth usrname. Defaults to None.
+            password (Optional[str], optional): kibana auth password. Defaults to None.
```

Comment:
Default username and password to `admin` `admin` instead of `None`.

Otherwise,  the following calls will fail or require exception handling down the line.

```python3
652: rmsuccess = existing.uninstall(self.kibana_url, auth=(username, password), force=True)
```

```python3
662:  result = self.import_kibana_saved_objects(username=username, password=password,
663:                                                 kibana_objects_file=kibana_objects_file)
```

Reply:
I think this was the bug that happened during my demo.